src/@3rdweb-sdk/react/hooks/useEngine.ts

```diff
@@ -180,6 +180,106 @@ export function useEngineUpdateVersion() {
   });
 }
 
+export function useEngineRemoveFromDashboard() {
+  const { user } = useLoggedInUser();
+  const queryClient = useQueryClient();
+
+  return useMutation(
+    async (instanceId: string) => {
+      invariant(instanceId, "instance is required");
+
+      const res = await fetch(`${THIRDWEB_API_HOST}/v1/engine/${instanceId}`, {
+        method: "DELETE",
+        credentials: "include",
+      });
+      if (!res.ok) {
+        throw new Error(`Unexpected status ${res.status}`);
+      }
+      res.body?.cancel();
```

Comment:
we *always* cancel the response body here? this seems odd, no? we should only cancel the response body if `!res.ok` I think before we throw?

Reply:
I meant to use the body in the throw which is why this `body.cancel()` is only if `res.ok` is truthy. I don't need the body from these responses (I assume 2xx is good).

Reply:
I believe you have to return *something* from the mutation function or else react query gets unhappy, so just return true or the res or something there, yeah (also no need to cancel the body in that case)